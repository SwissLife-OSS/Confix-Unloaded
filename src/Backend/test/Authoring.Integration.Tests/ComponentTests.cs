using Confix.Authentication.Authorization;
using Squadron;
using Snapshooter;
using Xunit;
using static Confix.Authoring.Integration.Tests.Wellknown;
using Snapshooter.Xunit;

namespace Confix.Authoring.Integration.Tests;

public class ComponentTests : IClassFixture<MongoResource>
{
    private readonly MongoResource _mongoResource;

    public ComponentTests(MongoResource mongoResource)
    {
        _mongoResource = mongoResource;
    }

    [Fact]
    public async Task Components_Should_ReturnAllComponents()
    {
        // arrange
        await TestExecutorBuilder
            .New()
            .AddDatabase(_mongoResource.ConnectionString)
            .AddDefaultUser()
            .AddPermission(Namespaces.Default, Scope.Component, Permissions.Read)
            .AddClient(out var client)
            .SetupDb(db => db
                .AddComponent(x => x with { Id = Guid.Parse("10000000-0000-0000-0000-000000000000"), Name = "Component 1" })
                .AddComponent(x => x with { Id = Guid.Parse("20000000-0000-0000-0000-000000000000"), Name = "Component 2" }))
            .Build();

        // act
        var result = await client.Value.GetComponents.ExecuteAsync();

        // assert
        result.AssertNoErrors();
        result.Data.MatchSnapshot();
    }
}
