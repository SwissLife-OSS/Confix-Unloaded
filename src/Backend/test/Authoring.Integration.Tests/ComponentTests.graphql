query GetComponents {
  components {
    nodes {
      ...Component
    }
  }
}
