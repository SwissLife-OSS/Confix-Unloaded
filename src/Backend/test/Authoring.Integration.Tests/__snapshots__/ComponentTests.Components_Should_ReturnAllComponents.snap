{
  "Components": {
    "Nodes": [
      {
        "Id": "Q29tcG9uZW50CmcxMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMA==",
        "Name": "Component 1",
        "Version": 1,
        "Schema": {
          "RootElement": {
            "ValueKind": "Array"
          }
        },
        "SchemaSdl": "type Configuration {\n    field: String!\n    nestedField: NestedConfiguration!\n}\ntype NestedConfiguration {\n    field: String!\n}",
        "Defaults": {
          "RootElement": {
            "ValueKind": "Object"
          }
        },
        "Values": {
          "RootElement": {
            "ValueKind": "Object"
          }
        },
        "State": "Active",
        "SchemaViolations": [],
        "Scopes": [
          {
            "Application": null,
            "Namespace": "Test",
            "ApplicationPart": null
          }
        ]
      },
      {
        "Id": "Q29tcG9uZW50CmcyMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMA==",
        "Name": "Component 2",
        "Version": 1,
        "Schema": {
          "RootElement": {
            "ValueKind": "Array"
          }
        },
        "SchemaSdl": "type Configuration {\n    field: String!\n    nestedField: NestedConfiguration!\n}\ntype NestedConfiguration {\n    field: String!\n}",
        "Defaults": {
          "RootElement": {
            "ValueKind": "Object"
          }
        },
        "Values": {
          "RootElement": {
            "ValueKind": "Object"
          }
        },
        "State": "Active",
        "SchemaViolations": [],
        "Scopes": [
          {
            "Application": null,
            "Namespace": "Test",
            "ApplicationPart": null
          }
        ]
      }
    ]
  }
}
